### PR TITLE
Move recent files to a separate submenu

### PR DESCRIPTION
**Pull request by franciszek-koltuniuk-red**
Changes:
 - move the list of recent files to a dedicated submenu
 - added an option to configure the number of recent files to list
 - added option to clean up the recent files list

### CODE REVIEW

**Comment by variar** [comment]
Thanks! I'll check this out

File: src/ui/src/recentfiles.cpp [hunk 1]
```diff
@@ -45,10 +51,27 @@ void RecentFiles::addRecent( const QString& text )
...
+}
+
+int RecentFiles::filesHistoryMaxItems() const
+{
+    return filesHistoryMaxItems_;
+}
+
+void RecentFiles::setFilesHistoryMaxItems( const int recentMaxFiles )
+{
+    if ( recentMaxFiles > 0 && recentMaxFiles <= MAX_RECENT_FILES ) {
```
Comment (variar) [inline src/ui/src/recentfiles.cpp:70]: this could be `std::clamp`

File: src/ui/src/recentfiles.cpp [hunk 2]
```diff
@@ -71,19 +94,21 @@ void RecentFiles::saveToStorage( QSettings& settings ) const
...
     LOG_DEBUG << "RecentFiles::retrieveFromStorage";
 
-    recentFiles_.clear();
+    removeAll();
 
     if ( settings.contains( "RecentFiles/version" ) ) {
         settings.beginGroup( "RecentFiles" );
         if ( settings.value( "version" ).toInt() == RECENTFILES_VERSION ) {
             int size = settings.beginReadArray( "filesHistory" );
+            size = ( size < MAX_RECENT_FILES ) ? size : MAX_RECENT_FILES ;
```
Comment (variar) [inline src/ui/src/recentfiles.cpp:111]: this would read better with  `std::min`

File: src/ui/src/recentfiles.cpp [hunk 3]
```diff
@@ -45,10 +51,27 @@ void RecentFiles::addRecent( const QString& text )
     recentFiles_.push_front( text );
 
     // Trim the list if it's too long
-    while ( recentFiles_.size() > MAX_NUMBER_OF_FILES )
+    while ( recentFiles_.size() > MAX_RECENT_FILES )
```
Comment (variar) [inline src/ui/src/recentfiles.cpp:54]: shouldn't this be trimmed to `filesHistoryMaxItems`?

File: src/ui/src/recentfiles.cpp [hunk 4]
```diff
@@ -45,10 +51,27 @@ void RecentFiles::addRecent( const QString& text )
...
         recentFiles_.pop_back();
 }
 
+int RecentFiles::getNumberItemsToShow() const
+{
+    return ( filesHistoryMaxItems_ < recentFiles_.count() ? filesHistoryMaxItems_ : recentFiles_.count() );
```
Comment (variar) [inline src/ui/src/recentfiles.cpp:60]: `std::min(recentFiles_.count(), filesHistoryMaxItems_)` better communicates intention here

**Review by variar — COMMENTED** [review]
Thanks! Just a couple of minor changes and I'll merge this.

File: src/ui/src/recentfiles.cpp (hunk 1 above)
Comment (franciszek-koltuniuk-red) [reply src/ui/src/recentfiles.cpp]: updated

File: src/ui/src/recentfiles.cpp (hunk 2 above)
Comment (franciszek-koltuniuk-red) [reply src/ui/src/recentfiles.cpp]: updated

File: src/ui/src/recentfiles.cpp (hunk 4 above)
Comment (franciszek-koltuniuk-red) [reply src/ui/src/recentfiles.cpp]: updated

**Comment by franciszek-koltuniuk-red** [comment]
@variar please review last commit. 
Thanks

**Pull request merged**